src/nodes/ParagraphView.vue

```diff
@@ -0,0 +1,94 @@
+<!--
+  - @copyright Copyright (c) 2019 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<NodeViewWrapper class="vue-component">
+		<NodeViewContent class="content" />
+		<ReferenceList v-if="text !== null" :text="text" :limit="1" />
+	</NodeViewWrapper>
+</template>
+
+<script>
+import { NodeViewContent, nodeViewProps, NodeViewWrapper } from '@tiptap/vue-2'
+import { ReferenceList } from '@nextcloud/vue-richtext'
+import debounce from 'lodash.debounce'
+
+export default {
+	name: 'ParagraphView',
+	components: {
+		NodeViewWrapper,
+		NodeViewContent,
+		ReferenceList,
+	},
+	props: nodeViewProps,
+	data() {
+		return {
+			text: null,
+		}
+	},
+	watch: {
+		node: {
+			handler(newNode) {
+				this.debouncedUpdateText(newNode)
+			},
+		},
+	},
+	created() {
+		this.debouncedUpdateText = debounce((newNode) => {
+			this.text = this.getTextReference(newNode?.textContent)
+		}, 500)
+		this.text = this.getTextReference(this.node?.textContent)
+	},
+	beforeUnmount() {
+		this.debouncedWatch?.cancel()
+	},
+	methods: {
+		getTextReference(text) {
+			const PATTERN = /(^)(https?:\/\/)?((?:[-A-Z0-9+_]+\.)+[-A-Z]+(?:\/[-A-Z0-9+&@#%?=~_|!:,.;()]*)*)($)/ig
+			if ((new RegExp(PATTERN)).test(text)) {
+				return text
+			}
+
+			return null
+		},
+	},
+}
+</script>
+<style lang="scss" scoped>
+.widgets--list {
+	margin-top: -20px;
+}
+:deep(div.widgets--list a.widget-default) {
+	color: var(--color-main-text);
+	padding: 0;
+	text-decoration: none;
+}
+:deep(.widget-default--details) {
+	overflow:hidden;
+	p {
+		margin-bottom: 4px !important;
+	}
+}
+</style>
+<style lang="scss">
+@import '@nextcloud/vue-richtext/dist/style.css'
+</style>
```

Comment:
We can import it direct in script statement

```html
<script>
import "@nextcloud/vue-richtext/dist/style.css"
</script>
``` 
Or using src attribute. 
https://v2.vuejs.org/v2/guide/single-file-components.html#What-About-Separation-of-Concerns

```html
<style src="@nextcloud/vue-richtext/dist/style.css"></style>
``` 

Also, we do not need `lang="scss"`